app/src/main/java/is/genki/bigredapp/android/DiningLocationActivity.java

```diff
@@ -0,0 +1,269 @@
+package is.genki.bigredapp.android;
+
+import android.support.annotation.Nullable;
+import android.support.v7.app.ActionBarActivity;
+import android.support.v4.app.Fragment;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DiningLocationActivity extends ActionBarActivity {
+
+    public static final String KEY_DINING_HALL = "DiningLocationActivity.DINING_HALL";
+    public static final String KEY_DINING_HALL_URL = "DiningLocationActivity.DINING_HALL_URL";
+
+    private static final String KEY_MEALS = "DiningLocationActivity.MEALS";
+    private static final String KEY_MENUS = "DiningLocationActivity.MENUS";
+    private static final String KEY_FRAGMENT = "DiningLocationActivity.FRAGMENT";
+
+    private String mDiningHall;
+    private String mDiningHallUrl;
+    private PlaceholderFragment mFragment;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_dining_location);
+        mFragment = new PlaceholderFragment();
+        if (savedInstanceState == null) {
+            getSupportFragmentManager().beginTransaction()
+                    .add(R.id.container, mFragment)
+                    .commit();
+
+            Bundle extras = getIntent().getExtras();
+            if (extras != null) {
+                final String diningHall = extras.getString(KEY_DINING_HALL);
+                String url = extras.getString(KEY_DINING_HALL_URL);
+
+                setTitle(diningHall);
+                mDiningHall = diningHall;
+                mDiningHallUrl = url;
+
+                if (GetRequest.isConnected(this)) {
+                    // Async Task to get the menu for a dining hall
+                    new GetRequest() {
+                        @Override
+                        protected void onPostExecute(String result) {
+                            try {
+                                List<MealMenu> menus = new ArrayList<>();
+                                if (result == null) throw new JSONException("No result");
```

Comment:
You're totally right. That case I found was when the request timed out (e.g. for Jansen's Dining Room request, which is currently broken). Sorry about that! Reverted in 8a99992faf339fa5c04b517c21d47566e280a089